src/App.tsx

```diff
@@ -1,26 +1,62 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { useDispatch } from 'react-redux';
+import { todosSlice } from './features/todos';
+import { useAppSelector } from './app/hooks';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+  const dispatch = useDispatch();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(todosFromServer => {
+        dispatch(todosSlice.actions.saveTodos(todosFromServer));
+
+        if (todosFromServer.length === 0) {
+          setError('There are no todos yet.');
+        }
+      })
+      .catch(() => {
+        setError('Something went wrong.');
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
+
+            <div className="block">
+              {loading && <Loader />}
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+              {error ? (
+                <p className="notification is-warning">{error}</p>
+              ) : (
+                <TodoList />
+              )}
```

Comment:
Consider thinking about this ternary. Because right now if you don't have any errors you will show list of todos even if you don't have any todos or have loading status